src/data/option/n_ary.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.option.basic
+
+/-!
+# Binary map of options
+
+This file defines the binary map of `option`. This is mostly useful to define pointwise operations
+on intervals.
+
+## Main declarations
+
+* `option.map₂`: Binary map of options.
+
+## Notes
+
+This file is very similar to the n-ary section of `data.set.basic`, to `data.finset.n_ary` and to
+`order.filter.n_ary`. Please keep them in sync.
+
+We do not define `option.map₃` as its only purpose would be to prove properties of `option.map₂`
```

Comment:
I'm not sure I believe this claim - what if I want to take the ternary image of a function under `option`?

Reply:
When you want to do that, we can review it, but if you look at the uses of `set.image3` and `filter.map₃` you will see that they are only used to prove associativity of the binary ones.

I mean if you want I can add "so far" but I really do not expect any more use because many operations are naturally binary (`+`,`-`,  `*`, `/`, `^`, `•`, `+ᵥ`, `-ᵥ`) while almost none is ternary (scalar triple product, I guess?).